js/script.js

```diff
@@ -1,60 +1,69 @@
-  bookDiv.innerHTML += bookElement;
-  bookDiv.querySelector('.removeBtn').addEventListener('click', (event) => {
-    const book = document.getElementById(event.target.parentElement.id);
-    book.parentElement.removeChild(book);
-    let storedData = JSON.parse(localStorage.getItem('storedData'));
-    storedData = storedData.filter((bookobj) => {
-      if (bookobj.id.toString() === book.id.substring(5, book.id.length)) {
-        return false;
-      }
-      return true;
+    bookDiv.innerHTML += bookElement;
+    bookDiv.querySelector('.removeBtn').addEventListener('click', (event) => {
+      const book = document.getElementById(event.target.parentElement.id);
+      this.booksList.splice(event.target.parentElement.id, 1);
+      book.parentElement.removeChild(book);
+      let storedData = JSON.parse(localStorage.getItem('storedData'));
+      storedData = storedData.filter((bookobj) => {
+        if (bookobj.id.toString() === book.id.substring(5, book.id.length)) {
+          return false;
+        }
+        return true;
+      });
+      localStorage.setItem('storedData', JSON.stringify(storedData));
     });
-    localStorage.setItem('storedData', JSON.stringify(storedData));
-  });
-  booksCont.append(bookDiv);
-}
+    booksCont.append(bookDiv);
+    this.booksList.push(book);
+  }
 
-function storeInputData() {
-  const inputData = {
-    title: document.getElementById('title').value,
-    author: document.getElementById('author').value,
-    id: Math.floor(Math.random() * 10000),
-  };
-  const storedData = JSON.parse(localStorage.getItem('storedData'));
-  storedData.push(inputData);
-  localStorage.setItem('storedData', JSON.stringify(storedData));
-  createBook(inputData);
-}
+  storeInputData() {
+    const inputData = {
+      title: document.getElementById('title').value,
+      author: document.getElementById('author').value,
+      id: Math.floor(Math.random() * 10000),
+    };
+    const storedData = JSON.parse(localStorage.getItem('storedData'));
+    storedData.push(inputData);
+    localStorage.setItem('storedData', JSON.stringify(storedData));
+    this.createBook(inputData);
+  }
 
-const initLoad = (booksList) => {
-  const storedData = JSON.parse(localStorage.getItem('storedData'));
-  if (storedData) {
-    storedData.forEach((book) => {
-      createBook(book);
-    });
-  } else {
-    localStorage.setItem('storedData', JSON.stringify(booksList));
-    booksList.forEach((book) => {
-      createBook(book);
+  initLoad(booksList) {
+    const storedData = JSON.parse(localStorage.getItem('storedData'));
+    if (storedData) {
+      storedData.forEach((book) => {
+        this.createBook(book);
+      });
+    } else {
+      localStorage.setItem('storedData', JSON.stringify(booksList));
+      booksList.forEach((book) => {
+        this.createBook(book);
+      });
+    }
+    const addBtn = document.getElementById('addBtn');
+    addBtn.addEventListener('click', () => {
+      this.storeInputData();
     });
```

Comment:
- [ OPTIONAL] After adding a book, I would advise you to clear the input fields for the title and author. This will give a better user experience for your application. 💻

Reply:
@Gambit142 okay we will add clear input functionalities